File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hashmap={}
        threshold=len(nums)//2

        for num in nums:
            hashmap[num]=hashmap.get(num,0)+1
            if hashmap[num]>threshold:
                return num
        return -1

File: 169-majority-element/majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
       int candidate=0;
       int count=0;

        foreach ( int item in nums)
        {
            if (count==0){
                candidate=item;
            }
            if(item==candidate){
                count+=1;
            }
            else
            {
                count-=1;
            }
        }
        return candidate;
        
    }
}